### ISSUE TITLE
waiting area abnormal parking

### ISSUE THREAD

**Issue by ZhenYu112** [eclipse-sumo/sumo#15565]
Add your issue description here.

*If possible, upload an example that shows your problem.*

**SUMO-version:**1.20.0

**operating system:**windows

I want to use the linkindex and linkindex2 attributes in the connection to implement the waiting area function in Sumo for secondary clearance, but I found that the vehicle will stop once at the parking line when the green light is cleared (such as going straight south to north in the picture below). I want to know why this abnormal parking phenomenon occurs? Hope you have time to answer, thank you very much!
![j1a1a-fxnvw](https://github.com/user-attachments/assets/98bee2f5-4553-4736-bc8a-a38e983dce8f)

**Comment by namdre** [comment]
the observed behavior is unrelated to the waiting area. Instead it appears to a be a bug related to cooperative lane changing where the vehicle briefly slows down to aid the queued vehicle in lane index 2.  The front vehicle in lane 2 must change lane to continue its left turn movement (via turning lane with index 2) which triggers cooperative behavior (which is not helping, hence a bug).  As a work-around avoid insertion turning vehicles on the wrong lane (i.e. by setting vehicle attribute `departLane="best"`)

**Comment by ZhenYu112** [comment]
> the observed behavior is unrelated to the waiting area. Instead it appears to a be a bug related to cooperative lane changing where the vehicle briefly slows down to aid the queued vehicle in lane index 2. The front vehicle in lane 2 must change lane to continue its left turn movement (via turning lane with index 2) which triggers cooperative behavior (which is not helping, hence a bug). As a work-around avoid insertion turning vehicles on the wrong lane (i.e. by setting vehicle attribute `departLane="best"`)

Thank you very much for your answer. I think I may not have expressed myself clearly. What I want to say is the position pointed by the arrow in the following picture, not the departure position. The 'linkindex' I mentioned is the 'linkindex' in the second picture below. I'm not sure if I expressed it clearly.
![企业微信截图_17285482562452](https://github.com/user-attachments/assets/61ffa82b-0588-44bf-b41e-210fc1832a37)
![企业微信截图_17285486188657](https://github.com/user-attachments/assets/d6b1b1b3-4568-4e0e-82b2-bbf539657dda)

**Comment by m-kro** [comment]
@ZhenYu112 Do you mean you want a second signal-controlled stop line within the same intersection?

**Comment by ZhenYu112** [comment]
> @ZhenYu112 Do you mean you want a second signal-controlled stop line within the same intersection?

Yes, in the GIF image above, you can see that left turning vehicles will stop in the middle of the intersection. In fact, there is also a signal control for straight ahead at this time, which is in a green light state

**Comment by namdre** [comment]
I know see what you mean and I don't have an explanation for it. Please attach the simulation files for closer inspection of the situation.

**Comment by ZhenYu112** [comment]
> I know see what you mean and I don't have an explanation for it. Please attach the simulation files for closer inspection of the situation.
Okay, this is the sumo file I used
[test0.zip](https://github.com/user-attachments/files/17325740/test0.zip)

**Comment by namdre** [comment]
the problem is caused by setting the connection attribute 'indirect'. This is only valid for indirect bicycle turns and netconvert even gives the warning `Could not compute indirect left turn shape`
You can still use linkIndex2 even if indirect=false.
